### PR TITLE
Fix display for literal members

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Fixes https://github.com/Microsoft/TypeScript/issues/606#issuecomment-143290638

A picture of the old behavior for convenience:
image

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -2078,7 +2106,7 @@ namespace ts {
                     }
                 }
 
-                function writeTypeofSymbol(type: ObjectType, typeFormatFlags?: TypeFormatFlags) {
+                function writeTypeOSymbol(type: ObjectType, typeFormatFlags?: TypeFormatFlags) {
```
Comment (weswigham) [inline src/compiler/checker.ts]: I don't think you meant to rename this function like this. :smile:

File: tests/cases/fourslash/quickInfoDisplayPartsEnum2.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    function verifyEnumDisplay() {
+        marker++;
+        goTo.marker(marker.toString());
+        verify.verifyQuickInfoDisplayParts("enum", "", { start: test.markerByName(marker.toString()).position, length: enumName.length },
+            (isConst ? [{ text: "const", kind: "keyword" }, { text: " ", kind: "space" }] : []).concat(
+                [{ text: "enum", kind: "keyword" }, { text: " ", kind: "space" }, { text: enumName, kind: "enumName" }]),
+            []);;
+    }
+
+    function verifyEnumMemberDisplay(enumMemberName: string, spanLength: number, initializer: number) {
```
Comment (weswigham) [inline tests/cases/fourslash/quickInfoDisplayPartsEnum2.ts]: Given that you reuse logic similar to this in three tests, maybe this kind of punctuated quick info verification should be added to the verifiers in `fourslash.ts`?

File: src/compiler/checker.ts (hunk 1 above)
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: I couldn't use an apostrophe. 😉

File: src/compiler/checker.ts [hunk 3]
```diff
@@ -1808,13 +1808,39 @@ namespace ts {
...
+             * Writes a property access or element access with the name of the symbol out to the writer.
+             * Uses the original source text for the name of the symbol if it is available to match how the user wrote the name,
+             * ensuring that any names written with literals use element accesses.
+             */
+            function appendPropertyOrElementAccessForSymbol(symbol: Symbol, writer: SymbolWriter): void {
+                const symbolName = getNameOfSymbol(symbol);
+                const firstChar = symbolName.charCodeAt(0);
+                const needsElementAccess = !isIdentifierStart(firstChar, languageVersion);
+
+                if (needsElementAccess) {
```
Comment (sheetalkamat) [inline src/compiler/checker.ts]: nice 👍

**Comment by sheetalkamat** [comment]
👍

File: tests/cases/fourslash/quickInfoDisplayPartsEnum2.ts (hunk 2 above)
Comment (DanielRosenwasser) [inline tests/cases/fourslash/quickInfoDisplayPartsEnum2.ts]: That's not a bad idea. I think if I run into more of these in the future I'll try to consolidate them.

**Pull request merged**